ansible/files/archives/opt-custom/etc/rc-pre-network.d/010-ipf-restore.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash -e
+#
+# This script generates firewall rules from files
+# stored (most likely in) /opt/custom/etc/ipf.d
+# It searches for files starting with the name
+# of the respective conf file, e.g. for ipf.conf
+# It matches either ipf.conf or ipf.conf-001, etc.
+#
+# It is intended for two things:
+# 1. restore fw config from a permanent storage
+# 2. split the fw config into multiple files
+#
+# Recommended format is <confname>-NNN
+# Examples: 
+# ipf.conf-005 ipf.conf-010 ...
+# ipnat.conf-005 ipnat.conf-010 ...
+# ipf6.conf-005 ipf6.conf-010 ...
+#
+# If no matching files exist, the firewall
+# config is not touched.
+#
+
+RULES_DIR="$(realpath "$(dirname $0)/../../etc/ipf.d")"
+GENERATE_FILES="ipf.conf ipnat.conf ipf6.conf"
+OUTPUT_DIR="/etc/ipf"
+IPF_SERVICE="svc:/network/ipfilter:default"
+
+. /lib/svc/share/smf_include.sh
+
+update_fw() {
+	if [[ ! -d "${RULES_DIR}" ]]; then
+		# nothing to do
+		exit $SMF_EXIT_OK
+	fi
+
+	FW_MODIFIED=0
+	for CFGFILE in ${GENERATE_FILES}; do
+		# get list of all conf files in RULES_DIR
+		# that end with requested filename (e.g. ipf.conf*)
+		# and sort them by name before merging
+		set +e
+		CONF_LIST="$(ls -1 ${RULES_DIR}/${CFGFILE}* 2> /dev/null)"
+		set -e
+
+		if [[ -z "${CONF_LIST}" ]]; then
+			# nothing to do
+			continue
+		fi
+
+		echo "Generating firewall rules for ${CFGFILE}"
+		echo "${CONF_LIST}" | xargs cat > "${OUTPUT_DIR}/${CFGFILE}"
```

Comment:
Hmm `cat ${CONF_LIST}` would do the same job, no?

Reply:
AFAIK no. The problem is sorting.

Reply:
Could you please leave a comment in the code so we don't forget about this.